src/sbvr-api/sbvr-utils.ts

```diff
@@ -330,6 +330,32 @@ const prettifyConstraintError = (
 	}
 };
 
+const cachedIsModelNew: { hasRun: boolean; models: Set<string> | undefined } = {
+	hasRun: false,
+	models: undefined,
+};
+
+export const isModelNew = async (
+	tx: Db.Tx,
+	modelName: string,
+): Promise<boolean> => {
+	const result = await tx.tableList("name = 'model'");
+	if (result.rows.length === 0) {
+		return false;
+	}
+	if (!cachedIsModelNew.hasRun) {
+		const { rows } = await tx.executeSql(
+			`SELECT "is of-vocabulary" FROM "model";`,
```

Comment:
Since you're switching to checking the `model` resource for existence is it now possible that the model could be created but then an error happen such that the `model` entry is populated but the `migrations` entry is not?

Reply:
Specifically I was thinking if it's possible for the `INSERT INTO "model" ...` to have been committed and then things fail so `isModelNew(...)` would return `false` but the migrations table wouldn't have an entry and therefore it would try to run all the historical migrations on a new model. I realized though this might not actually be a problem though as long as it's all run in the same transaction?

Reply:
Ah I see. I'll double check this and assume that this will be handled in a single transaction procedure that will fail.
Although using the method should be usable across different transactions for the usage of sync and async migration execution (which will have different transactions)

Reply:
@Page-

From the execution order the `isModelNew(...)` right now is executed before the `INSERT INTO "model"` so that the database is evaluated as it was before starting up pine (before calling config-loader).

I've double checked that the isModelNew(...) query needs to run inside the:
https://github.com/balena-io/pinejs/blob/7789e2a9e3da739b33f4b2250527782696255bbc/src/config-loader/config-loader.ts#L93-L94 config-loader transaction. 
Do you agree on leaving a comment on the isModelNew so that it needs to be executed in one same transaction?

Reply:
I think that's probably the way to go